### PR TITLE
COVID-19 India: Added district-level daily case counts

### PR DESCRIPTION
**Pull request by SenthamizhanV**
@pradh @thejeshgn 
Commit contains district-level COVID-19 case counts for India. Districts mapped with Local Govt Directory IDs. Retained the schema from states data.

### CODE REVIEW

File: scripts/covid19_india/cases_count_districts_data/test_data/expected.csv [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Date,District,districtCode,Count_MedicalTest_ConditionCOVID_19_Positive,Count_MedicalConditionIncident_COVID_19_PatientRecovered,Count_MedicalConditionIncident_COVID_19_PatientDeceased
+26-04-2020,Anantapur,502.0,53,14,4
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/test_data/expected.csv:2]: District code should be a 3 digit string

File: scripts/covid19_india/cases_count_districts_data/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+[MoHFW](https://www.mohfw.gov.in/) publishes cumulative confirmed, cured, death due to COVID-19 counts, everyday. It's published for every State and Union Territory. Sometimes it's published more than once a day. It's scraped by the [covid19india](https://covid19india.org).
+
+
+#### Cleaned Data
+- [COVID19_cases_indian_states.csv](COVID19_cases_indian_states.csv).
+
+#### Template MCFs
+- [COVID19_cases_indian_states.tmcf](COVID19_cases_indian_states.tmcf).
+
+
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/README.md:21]: should this part talk about district data/files?

Also, please indicate how we plan to resolve the places. using indianCensusAreaCode2011 property?

File: scripts/covid19_india/cases_count_districts_data/COVID19_cases_indian_districts.tmcf [hunk 3]
```diff
@@ -0,0 +1,30 @@
+Node: E:COVID19_cases_india->E0
+typeOf: dcs:StatVarObservation
+variableMeasured: dcs:Count_MedicalTest_ConditionCOVID_19_Positive
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/COVID19_cases_indian_districts.tmcf:3]: If these are cumulative counts, per README, I think the three stat-vars should be:
* CumulativeCount_MedicalTest_ConditionCOVID_19_Positive
* CumulativeCount_MedicalConditionIncident_COVID_19_PatientRecovered
* CumulativeCount_MedicalConditionIncident_COVID_19_PatientDeceased

File: scripts/covid19_india/cases_count_districts_data/preprocess.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            'Recovered':
+                'Count_MedicalConditionIncident_COVID_19_PatientRecovered'
+        })
+    df = df[output_columns]
+    df.to_csv(csv_file_path, index=False, header=True)
+
+
+if __name__ == '__main__':
+    api_url = "https://api.covid19india.org/csv/latest/districts.csv"
+    data = pd.read_csv(api_url)
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/preprocess.py:73]: nit: df

File: scripts/covid19_india/cases_count_districts_data/preprocess.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+]
+
+
+def _get_district_code(district):
+    dist_code = pd.read_csv('../common/data/districts-local-directory.csv')
+    if district not in ['Unknown']:
+        close_match = difflib.get_close_matches(str(district).upper(), dist_code['DistrictName(InEnglish)'],
+                            n=1, cutoff=0.8)
+        if close_match:
+            return dist_code[dist_code['DistrictName(InEnglish)'] == close_match[0]]['DistrictCode'].values[0]
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/preprocess.py:39]: comment what this does. and if you manually inspect the results to see how good/bad the match was?

File: scripts/covid19_india/cases_count_districts_data/preprocess.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    # creating dictionary with district vs code mappings
+    district_codes = []
+    
+    for district in df['District'].unique():
+        district_codes.append(_get_district_code(district))
+    dist_codes_dict = dict(zip(df['District'].unique().tolist(), district_codes))
+    
+    #get the district codes and drop rows that do not have district codes
+    df['districtCode'] = df.apply(lambda row: dist_codes_dict[row['District']],
+                                                axis=1)
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/preprocess.py:54]: how many do we drop?

File: scripts/covid19_india/cases_count_districts_data/preprocess.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+"""
+Retrieving and preprocessing district-wise daily case counts for India.
+Raw data retrieved using API by covid19india.org (https://www.covid19india.org/)
+"""
+
+
+output_columns = [
+    'Date', 'District', 'districtCode',
+    'Count_MedicalTest_ConditionCOVID_19_Positive',
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/preprocess.py:27]: might need updating per the new stat-var suggestions

File: scripts/covid19_india/cases_count_districts_data/test_data/test_data.csv [hunk 8]
```diff
@@ -0,0 +1,20 @@
+Date,State,District,Confirmed,Recovered,Deceased,Other,Tested
+26-04-2020,Andaman and Nicobar Islands,Unknown,33,11,0,0,
```
Comment (pradh) [inline scripts/covid19_india/cases_count_districts_data/test_data/test_data.csv:2]: Consider demoing this using the previewer too.

https://datacommonsorg.github.io/tools/mcf-browser/#&file=https://raw.githubusercontent.com/datacommonsorg/data/2a6de8fcfd1584c672402229698d93d03e67ef45/scripts/india_lgd/local_government_directory_states/LocalGovernmentDirectory_States.tmcf&file=https://raw.githubusercontent.com/datacommonsorg/data/2a6de8fcfd1584c672402229698d93d03e67ef45/scripts/india_lgd/local_government_directory_states/LocalGovernmentDirectory_States.csv

File: scripts/covid19_india/cases_count_districts_data/test_data/expected.csv (hunk 1 above)
Comment (SenthamizhanV) [reply scripts/covid19_india/cases_count_districts_data/test_data/expected.csv]: Converted it to string. But couldn't retain codes like '079'. It gets changes to '79' but that is how it is in Thej's LGD PR. So I retained it.

File: scripts/covid19_india/cases_count_districts_data/COVID19_cases_indian_districts.tmcf (hunk 3 above)
Comment (SenthamizhanV) [reply scripts/covid19_india/cases_count_districts_data/COVID19_cases_indian_districts.tmcf:3]: These are daily counts. Not cumulative counts. The README is wrong. I corrected it now.

File: scripts/covid19_india/cases_count_districts_data/preprocess.py (hunk 4 above)
Comment (SenthamizhanV) [reply scripts/covid19_india/cases_count_districts_data/preprocess.py]: Changed `data` to `df`

File: scripts/covid19_india/cases_count_districts_data/preprocess.py (hunk 5 above)
Comment (SenthamizhanV) [reply scripts/covid19_india/cases_count_districts_data/preprocess.py]: Added some comments for the function and `close_match` line. And I looked at difference cutoffs to see which accounts for all variations for the same name without any mismatch. 0.8 was the optimal one.

File: scripts/covid19_india/cases_count_districts_data/preprocess.py (hunk 6 above)
Comment (SenthamizhanV) [reply scripts/covid19_india/cases_count_districts_data/preprocess.py]: 61 districts from the covid19india.org data (643 -> 582). I added a comment in the code mentioning the same.

**Pull request merged**